src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1440,32 +1440,27 @@ public void deleteAllFeedbackSessionsCascade(List<InstructorAttributes> instruct
 
     /**
      * Soft-deletes a specific feedback session to Recycle Bin.
-     * @return Soft-deletion time of the feedback session.
+     * @return the time when the feedback session is moved to the recycle bin
      */
     public Instant moveFeedbackSessionToRecycleBin(String feedbackSessionName, String courseId)
-            throws InvalidParametersException, EntityDoesNotExistException {
-        FeedbackSessionAttributes feedbackSession = fsDb.getFeedbackSession(courseId, feedbackSessionName);
-        feedbackSession.setDeletedTime();
-        fsDb.updateFeedbackSession(feedbackSession);
+            throws EntityDoesNotExistException {
 
-        return feedbackSession.getDeletedTime();
+        return fsDb.softDeleteFeedbackSession(feedbackSessionName, courseId);
     }
 
     /**
      * Restores a specific feedback session from Recycle Bin to feedback sessions table.
      */
     public void restoreFeedbackSessionFromRecycleBin(String feedbackSessionName, String courseId)
-            throws InvalidParametersException, EntityDoesNotExistException {
-        FeedbackSessionAttributes feedbackSession = fsDb.getSoftDeletedFeedbackSession(courseId, feedbackSessionName);
-        feedbackSession.resetDeletedTime();
-        fsDb.updateFeedbackSession(feedbackSession);
+            throws EntityDoesNotExistException {
+        fsDb.restoreDeletedFeedbackSession(feedbackSessionName, courseId);
     }
 
     /**
      * Restores all feedback sessions from Recycle Bin to feedback sessions table.
```

Comment:
`feedback sessions table`?

Reply:
But this is in the logic layer?

Reply:
`feedback sessions table` sounds like UI-specific.